Gems/Terrain/Code/Source/TerrainRenderer/TerrainFeatureProcessor.cpp

```diff
@@ -162,9 +149,12 @@ namespace Terrain
             queryResolution, &AzFramework::Terrain::TerrainDataRequests::GetTerrainHeightQueryResolution);
         // Currently query resolution is multidimensional but the rendering system only supports this changing in one dimension.
 
-        m_terrainBounds = worldBounds;
-        m_sampleSpacing = queryResolution;
-        m_heightmapNeedsUpdate = true;
+        if (m_terrainBounds != worldBounds || m_sampleSpacing != queryResolution)
```

Comment:
Nit: The comment here can get removed at this point, since query resolution is no longer multidimensional.